test: add tests for HTTP API module

cristiand391 opened the pull request:
Adds unit tests for the HTTP API module (including SOAP wrapper).

UTs cover HTTP/SOAP headers, session refresh, parsing error in responses.

@W-14580746@

cristiand391 commented on karma.conf.js:
```diff
@@ -43,7 +43,7 @@ module.exports = function (config) {
 
     // list of files / patterns to exclude
-    exclude: [],
+    exclude: ['test/http-api.test.ts'],
```
`nock` only works in node so we skip this file in browser tests

cristiand391 commented on src/http-api.ts:
```diff
@@ -288,6 +288,11 @@ export class HttpApi<S extends Schema> extends EventEmitter {
+    // XML response
+    if (errors['Errors']) {
+      return errors['Errors']['Error'];
+    }
```
this cover errors in XML responses from the REST API, if you remove this block then the message in this test includes the whole XML instead of just the error props:
https://github.com/jsforce/jsforce/pull/1390/files#diff-de7739647545a012d3fb1a85782ad2a184f9b8803a7ffa84cedaf3774d58b82aR238

example using `org api`:
```
➜  jsforce git:(cd/http-module-tests) ✗ sf org api '/services/data/v59.0/sobjects/Broker__c/a008N0000032UmoQA' -o test-rkr5unx057se@example.com --header 'Accept:application/json'
[
  {
    "errorCode": "NOT_FOUND",
    "message": "Provided external ID field does not exist or is not accessible: a008N0000032UmoQA"
  }
]
➜  jsforce git:(cd/http-module-tests) ✗ sf org api '/services/data/v59.0/sobjects/Broker__c/a008N0000032UmoQA' -o test-rkr5unx057se@example.com --header 'Accept:application/xml'
<?xml version="1.0" encoding="UTF-8"?><Errors><Error><errorCode>NOT_FOUND</errorCode><message>Provided external ID field does not exist or is not accessible: a008N0000032UmoQA</message></Error></Errors>
```

cristiand391 commented on test/http-api.test.ts:
```diff
@@ -0,0 +1,431 @@
+
+      httpApi.on('request', (req: HttpRequest) => {
+        assert.ok(req?.headers?.['Authorization'] === `Bearer ${accessToken}`);
+        testPassed = true;
```
setting`testPassed` here ensures this code ran.

mshanemc commented on src/http-api.ts:
```diff
@@ -288,6 +288,11 @@ export class HttpApi<S extends Schema> extends EventEmitter {
+
+    // XML response
+    if (errors['Errors']) {
+      return errors['Errors']['Error'];
```
why the brackets and not not `errors.Errors` and `errors.Errors.Error`

cristiand391 replied:
no specific reason, made it use dot notation 👍🏼

mshanemc reviewed (approved):
potential test case ideas?

soap/rest responses are html content

cristiand391 commented:
@mshanemc I backported the HTML err handling from the 2.0 branch and added a UT for that, also added one more to cover the  `noContentResponse` option.
